src/convert.cpp

```diff
@@ -112,3 +112,64 @@ ddwaf_object* to_ddwaf_object(ddwaf_object *object, Napi::Env env, Napi::Value v
   // we use empty maps for now instead of null. See issue !43
   return ddwaf_object_map(object);
 }
+
+Napi::Value from_ddwaf_object(ddwaf_object *object, Napi::Env env) {
+  DDWAF_OBJ_TYPE type = object->type;
+
+  Napi::Value result;
+
+  switch (type) {
+    case DDWAF_OBJ_SIGNED:
+      result = Napi::Number::New(env, object->intValue);
+      break;
+    case DDWAF_OBJ_UNSIGNED:
+      result = Napi::Number::New(env, object->uintValue);
+      break;
+    case DDWAF_OBJ_STRING:
+      result = Napi::String::New(env, object->stringValue, object->nbEntries);
+      break;
+    case DDWAF_OBJ_ARRAY: {
+      Napi::Array arr = Napi::Array::New(env, object->nbEntries);
+
+      if (env.IsExceptionPending()) {
+        mlog("Exception pending");
+        return env.Null();
+      }
+
+      for (uint32_t i = 0; i < object->nbEntries; ++i) {
+        ddwaf_object e = object->array[i];
+        Napi::Value v = from_ddwaf_object(&e, env);
```

Comment:
Let's add a recursion limit just in case

Reply:
Just in case the waf gives me a recursive object reference ?

Reply:
yes